src/etos_suite_runner/lib/runner.py

```diff
@@ -49,33 +53,97 @@ def __init__(self, params, etos, context):
 
         self.context = context
 
-    def _run_etr_and_wait(self, environment):
-        """Run ETR based on number of IUTs and wait for them to finish.
+    def _release_environment(self, task_id):
+        """Release an environment from the environment provider.
+
+        :param task_id: Task ID to release.
+        :type task_id: str
+        """
+        wait_generator = self.etos.http.wait_for_request(
+            self.etos.debug.environment_provider, params={"release": task_id}
+        )
+        for response in wait_generator:
+            if response:
+                break
+
+    def _run_etr(self, environment):
+        """Trigger an instance of ETR.
 
         :param environment: Environment which to execute in.
         :type environment: dict
-        :return: List of test results from all ETR instances.
-        :rtype: list
         """
+        uri = environment["data"]["uri"]
+        json_header = {"Accept": "application/json"}
+        json_response = self.etos.http.wait_for_request(
+            uri,
+            headers=json_header,
+        )
+        suite = {}
+        for suite in json_response:
+            break
+        else:
+            raise Exception("Could not download sub suite instructions")
+
+        executor = Executor(self.etos)
+        executor.run_tests(suite)
+
+    def _environments(self, suite_name):
+        """Get environments for a specific test suite.
+
+        :param suite_name: Since environment defined events have a name starting with 'suite_name'
+                           we will use this as a part of getting the environments.
+        :type suite_name: str
+        :return: Test suite environments.
+        :rtype: iterator
+        """
+        yielded = []
+        status = {
+            "status": "FAILURE",
+            "error": "Couldn't collect any error information",
+        }
+        timeout = time.time() + self.etos.config.get("WAIT_FOR_ENVIRONMENT_TIMEOUT")
+        while time.time() < timeout:
+            time.sleep(1)
```

Comment:
This hammers the ER a bit too much. We should increase this sleep and also do one request per ESR instance instead of one request per suite.